napalm_nxos/nxos.py

```diff
@@ -67,8 +80,52 @@ def __init__(self, hostname, username, password, timeout=60, optional_args=None)
             self.port = optional_args.get('port', 443)
         elif self.transport == 'http':
             self.port = optional_args.get('port', 80)
+        elif self.transport == 'netmiko_nxos':
+            # Netmiko possible arguments
+            netmiko_argument_map = {
+                'port': None,
+                'verbose': False,
+                'global_delay_factor': 1,
+                'use_keys': False,
+                'key_file': None,
+                'ssh_strict': False,
+                'system_host_keys': False,
+                'alt_host_keys': False,
+                'alt_key_file': '',
+                'ssh_config_file': None,
+            }
 
-    def open(self):
+            fields = netmiko_version.split('.')
+            fields = [int(x) for x in fields]
+            maj_ver, min_ver, bug_fix = fields
+            if maj_ver >= 2:
+                netmiko_argument_map['allow_agent'] = False
+            elif maj_ver == 1 and min_ver >= 1:
```

Comment:
You can use `distutils.version.LooseVersion` and `SrictVersion` to simplify all this bunch of code.